awslabs/aws-media-insights-engine
# unit tests, simplified integ tests, e2e test, IAM auth for tests, bug…

… fix in apihandler role, new test stack format

*Issue #, if available:*

*Description of changes:*

Huge update to testing for the MIE framework.

- Local unit tests
- Simplified Integ test
- E2E test
- New testing directory structure
- Simplified test usage
- IAM auth for tests
- Simplified test resources 
- Various bug fixes; #190
- Testing Readme
- Testing coverage matrix
- Update xray patching to not patch modules if running locally
- Rename test stack to accurately reflect the contained resources
- Remove operator library conditional (setting it to false prohibits the stack from deploying at all)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
